### PR TITLE
fixed output ranges for split visc subfaces

### PR DESCRIPTION
**Pull request by joanibal**
## Purpose
The goal of this PR is to address is issue #157 
See the issue for details on the issue. 
TLDR: when the subfaces of viscous zones were split, the output data was slightly wrong 

## Type of change
- [x] Bugfix (non-breaking change which fixes an issue)

## Testing
For a problem with a surface output dependent on viscSubFace
1.   add the option "partitionLikeNProc": 50(may need to be higher if you have a finer mesh),
2.   try to write a surface solution file
3.   check for bad output or seg fault

## Checklist
_Put an `x` in the boxes that apply._

- [NA] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Comment by joanibal** [comment]
Before: 
Screenshot from 2021-08-16 14-31-35

After:
Screenshot from 2021-08-16 14-30-38

**Comment by anilyil** [comment]
Good stuff. I will go through the code and try to understand it a bit. Is there a way we can test this?

File: src/output/outputMod.F90 [hunk 1]
```diff
@@ -1881,25 +1884,42 @@ subroutine storeSurfsolInBuffer(sps, buffer, nn, blockID,   &
...
-                ii = min(i + offVis, rangeFace(1,2))
-             else if(i == rangeFace(1,2)) then
-                ii = max(i - offVis, rangeFace(1,1))
-             else
-                ii = i
-             endif
+         !  if satements are used to copy the value of of the interior
+         ! cell since the value isn't defined in the rind ell 
 
+         if (present(jBeg) .and. present(jEnd) .and. (useRindLayer)) then
```
Comment (anilyil) [inline src/output/outputMod.F90:1892]: Why did you define i/j beg/end as optional input variables here? Is this routine used anywhere else in adflow?

File: src/output/outputMod.F90 [hunk 2]
```diff
@@ -1881,25 +1884,42 @@ subroutine storeSurfsolInBuffer(sps, buffer, nn, blockID,   &
...
+         else
+            jj = j
+
+         end if
+
+          do i=rangeFace(1,1), rangeFace(1,2)
+             if (present(iBeg) .and. present( iEnd) .and. (useRindLayer)) then 
+               ior = i + iBegor - 1
+               if (ior == iBeg) then 
+                  ! print *, 'ibeg'
```
Comment (anilyil) [inline src/output/outputMod.F90:1910]: Can you remove the print statements?

File: src/output/writeCGNSSurface.F90 [hunk 3]
```diff
@@ -1320,7 +1320,8 @@ subroutine writeSurfaceSol
                call storeSurfsolInBuffer(ind, buffer, jj, ii,       &
                     faceID, cellRange(1,1,kk), &
                     solNames(mm),              &
-                    viscousSubface, storeRindLayer)
+                    viscousSubface, storeRindLayer,&
+                    iBeg, iEnd+1, jBeg, jEnd+1)
```
Comment (anilyil) [inline src/output/writeCGNSSurface.F90:1324]: Is there a reasoning (other than convenience) for adding the 1s here to i/j end here instead of the low level checks in the subroutine itself? The additions here can cause confusion; the i/j beg/end variables have the same names here and the subroutine you are calling here, but the values will be different. I would rather put the +1s right when the variables are actually used, and maintain the same values for these variables which have the same names. Just a convenience thing.

**Review by anilyil — CHANGES_REQUESTED** [review]
Thanks for fixing this, and also the detailed explanation in #157. I have left a few comments.
